cyipopt/scipy_interface.py

```diff
@@ -555,20 +550,50 @@ def minimize_ipopt(fun,
                           njev=problem.njev,
                           nit=problem.nit)
 
+
 def _minimize_ipopt_iv(fun, x0, args, kwargs, method, jac, hess, hessp,
                        bounds, constraints, tol, callback, options):
     # basic input validation for minimize_ipopt that is not included in
     # IpoptProblemWrapper
-
-    x0 = np.asarray(x0)[()]
+    x0 = np.atleast_1d(x0)
     if not np.issubdtype(x0.dtype, np.number):
         raise ValueError('`x0` must be a numeric array.')
 
     if method is not None:  # this will be updated when gh-200 is merged
         raise NotImplementedError('`method` is not yet supported.`')
 
-    # TODO: add input validation for `bounds` when adding
-    #  support for instances of new-style constraints (e.g. `Bounds`)
+    # Handle bounds that are either sequences (of sequences) or instances of
+    # `optimize.Bounds`.
+    if bounds is None:
+        bounds = [-np.inf, np.inf]
+
+    if isinstance(bounds, optimize.Bounds):
+        lb, ub = bounds.lb, bounds.ub
+    else:
+        bounds = np.atleast_2d(bounds)
+        if bounds.shape[1] != 2:
+            raise ValueError("`bounds` must specify both lower and upper "
+                             "limits for each decision variable.")
+        lb, ub = bounds.T
+
+    try:
+        lb, ub, x0 = np.broadcast_arrays(lb, ub, x0)
+    except ValueError:
+        raise ValueError("The number of lower bounds, upper bounds, and "
+                         "decision variables must be equal or broadcastable.")
+
+    try:
+        lb = lb.astype(np.float64)
+        ub = ub.astype(np.float64)
```

Comment:
Does IPOPT support other dtypes? 
I could try to maintain user floating point dtype here. The reason for this step is that if the user passes `bounds` as a list of lists with `None` to represent no limit / infinity, then `lb` and `ub` get converted to object arrays.

Reply:
The types are all set to double for these things:

https://github.com/mechmotum/cyipopt/blob/master/cyipopt/cython/ipopt_wrapper.pyx#L33

Reply:
OK, then this line should be fine.